spaceship/lib/spaceship/tunes/b2b_user.rb

```diff
@@ -17,8 +17,13 @@ class B2bUser < TunesBase
         'value.dsUsername' => :ds_username
       )
 
-      def self.from_username(username)
-        self.new({ 'value' => { 'add' => true, 'delete' => false, 'dsUsername' => username } })
+      def self.from_username(username, is_add_type: true)
+        self.new({ 'value' => { 'add' => is_add_type, 'delete' => !is_add_type, 'dsUsername' => username } })
+      end
+
+      # equality check for the two objects (for cleaning up some tests etc)
+      def ==(other)
```

Comment:
Is this **only** for tests like the comment? If so, this should probably be removed from here and put into the test file.

If not, does this mean a `josh@example.com` could both be "added" and "deleted" and not be seen as the same user? 🤔

Reply:
While this method is used only in test, it is still a method for this class. I can see future uses in the availability class but even if that weren't the case, I'd insist on this method being defined here. It would violate the object oriented design principle to not do it this way - what we can do here in one line ( and one more line in test ) would take nine lines in test. 

Well, yes. Since "add" and "delete" are class properties. And that is what we are checking in equality. So, despite them having the same email_id, a user to be added and a user to be removed are different by design.